src/pages/docs/guides/chat/build-embedded-chat.mdx

```diff
@@ -0,0 +1,414 @@
+
+// show who's typing
+room.typing.subscribe((event) => {
+  console.log('Currently typing:', event.currentlyTyping);
+});
+
+// attach the room
+await room.attach();
+
+// show a typing indicator (call on every keystroke, Ably Chat SDKs throttle this to a predefined interval)
+await room.typing.keystroke();
+
+// stop the typing indicator (for example when a message is sent)
+await room.typing.stop();
+```
+
+## Message reactions
+
+Message reactions are a ubiquitous feature in modern chat experiences and are a great way to enhance engagement and enable users to quickly express sentiment to key points in the conversation.
+
+Send a message reaction:
+<Code>
+```javascript
+const message; // your message
+await room.messages.reactions.send(message.serial, {name: '❤️'});
+```
+</Code>
+
+Message reactions in Ably Chat come in three types: `unique`, `distinct` and `multiple`, to suit different use cases: from one reaction per message to multiple reactions per message with or without counts. See the [Message reactions](/docs/chat/rooms/message-reactions) documentation for more details.
+
+## Message history: essential task context <a id="history"/>
+
+Message history is crucial for chat, ensuring continuity and context even when participants join mid-task or return after interruptions.
+
+Ably stores [chat history](/docs/chat/rooms/history) for 30 days by default, with options to extend up to a year on request.
+
+Message history provides users with:
+* **Task continuity:** New participants can quickly understand the current state and previous decisions.
+* **Context preservation:** Users returning to a task don't lose important information.
+* **Audit trail:** Complete conversation records for compliance, training, or dispute resolution.
+
+History is almost always beneficial:
+
+* **Support tickets:** Agents can see the full conversation history to understand the issue, after handover from automated/AI support or another agent.
+* **Collaborative work:** Team members can catch up on decisions and progress.
+* **Gaming sessions:** Players can review moves and strategy discussions.
+* **Marketplace transactions:** Complete communication record for orders and deliveries.
+
+<Code>
+```javascript
+// Get the chat room
+const room = await chatClient.rooms.get('support-ticket-123');
+
+// Subscribe to messages
+const subscription = room.messages.subscribe((messageEvent) => {
+  console.log('Received:', messageEvent);
+  // handle message event to update state and UI
+});
+
+// attach the room
+await room.attach();
+
+// Load recent history for context
+// This ensures you get a complete picture without missing messages
+let historyResponse = await subscription.historyBeforeSubscription({limit: 50});
+
+// Iterate through all pages
+while (true) {
+  console.log('Messages: ', historyResponse.items);
+  if (historyResponse.hasNext()) {
+    historyResponse = await historyResponse.next();
+  } else {
+    break;
+  }
+}
+```
+</Code>
+
+## React UI Kit
```

Comment:
I think we need some framing here too - what's the problem that UI kits are solving for this use case? Reduce time to build?